### ISSUE TITLE
Use formatting options per project and not globally

### ISSUE THREAD

**Issue by JulianH99** [fannheyward/coc-pyright#924]
I have a flask project where imports should not be sorted.
I solved this by setting
```
{
  "python.formatting.autopep8Args": ["--ignore", "E402"]
}
```
on `coc-settings.json`, but, this disables the import sorting globally, and I need to disable it only on the current flask project

**What's the output of `:CocCommand pyright.version`**

coc-pyright 1.1.304 with Pyright 1.1.304
**What's the output of `:CocCommand workspace.showOutput Pyright`**

Workspace: d:\Universidad\Semestre 13\virtus\projects\bps
Using python from C:\Users\Julian\AppData\Local\Programs\Python\Python310\python.EXE

[Info  - 20:41:38.866] Pyright language server 1.1.304 starting
[Info  - 20:41:38.866] Server root directory: C:\Users\Julian\AppData\Local\coc\extensions\node_modules\coc-pyright\node_modules\pyright/dist/
[Info  - 20:41:38.869] Starting service instance "bps"
[Info  - 20:41:38.882] Setting pythonPath for service "bps": "C:\Users\Julian\AppData\Local\Programs\Python\Python310\python.EXE"
[Info  - 20:41:38.882] Loading configuration file at d:\Universidad\Semestre 13\virtus\projects\bps\pyrightconfig.json
[Info  - 20:41:38.924] Assuming Python version 3.10
[Info  - 20:41:38.924] No include entries specified; assuming d:\Universidad\Semestre 13\virtus\projects\bps
[Info  - 20:41:38.927] Auto-excluding **/node_modules
[Info  - 20:41:38.927] Auto-excluding **/__pycache__
[Info  - 20:41:38.928] Auto-excluding **/.*
[Warn  - 20:41:38.928] stubPath d:\Universidad\Semestre 13\virtus\projects\bps\typings is not a valid directory.
[Info  - 20:41:39.040] Found 6 source files

**Comment by fannheyward** [comment]
https://github.com/fannheyward/coc-pyright/issues/923, use `:CocLocalConfig`.

**Comment by JulianH99** [comment]
It does not work. `:CocLocalConfig` shows my local config file under the `.vim` directory, but when I try and save my file it stills organizes imports

**Comment by fannheyward** [comment]
Can't reproduce, the configurations in `.vim/coc-settings.json` will be used first.

Check `:CocCommand workspace.showOutput coc-pyright-formatting`, autopep args will be displayed in logs.

**Comment by JulianH99** [comment]
Yeah, I executed the command and the autopep args are not being included, it's weird, I know it should work

this is the output
```
Using python from C:\Users\Julian\AppData\Local\Programs\Python\Python310\python.EXE

########## active formattor: autopep8
execPath:   C:\Users\Julian\AppData\Local\Programs\Python\Python310\Scripts\autopep8.EXE
moduleName: undefined
args:       --diff,d:\Universidad\Semestre 13\virtus\projects\bps\blog\app\main\__init__.py.5209f61f5350fd14ff65e1fc85e40e2a.py

########## autopep8 output:
--- original/d:\Universidad\Semestre 13\virtus\projects\bps\blog\app\main\__init__.py.5209f61f5350fd14ff65e1fc85e40e2a.py
+++ fixed/d:\Universidad\Semestre 13\virtus\projects\bps\blog\app\main\__init__.py.5209f61f5350fd14ff65e1fc85e40e2a.py
@@ -1,6 +1,4 @@
+from app.main import routes
 from flask import Blueprint
 
 bp = Blueprint("main", __name__)
-
-
-from app.main import routes

Using python from C:\Users\Julian\AppData\Local\Programs\Python\Python310\python.EXE

########## active formattor: autopep8
execPath:   C:\Users\Julian\AppData\Local\Programs\Python\Python310\Scripts\autopep8.EXE
moduleName: undefined
args:       --diff,d:\Universidad\Semestre 13\virtus\projects\bps\blog\app\main\__init__.py.5209f61f5350fd14ff65e1fc85e40e2a.py

########## autopep8 output:
--- original/d:\Universidad\Semestre 13\virtus\projects\bps\blog\app\main\__init__.py.5209f61f5350fd14ff65e1fc85e40e2a.py
+++ fixed/d:\Universidad\Semestre 13\virtus\projects\bps\blog\app\main\__init__.py.5209f61f5350fd14ff65e1fc85e40e2a.py
@@ -1,6 +1,4 @@
+from app.main import routes
 from flask import Blueprint
 
 bp = Blueprint("main", __name__)
-
-
-from app.main import routes
```

And this is my `coc-settings.json`
```
{
  "python.formatting.autopep8Args": ["--ignore", "E402"]
}

```

**Comment by fannheyward** [comment]
Still can't reproduce, here's my `.vim/coc-settigns.json`:

```json
{
  "python.formatting.autopep8Args": ["--ignore", "E402"]
}
```

used by formatting:

```
########## active formattor: autopep8
execPath:   /Users/fannheyward/src/hello-gradio/.venv/bin/autopep8
moduleName: undefined
args:       --diff,--ignore,E402,/Users/fannheyward/src/hello-gradio/foo.py.8824af40553904140a1dad194bcf16cb.py
```

Does `python.formatting.autopep8Args` work with in global coc-settings.json with your project?

**Comment by JulianH99** [comment]
@fannheyward I'll be trying this in a new project, just so you don't think I abandoned the issue in the meantime. Thanks!

**Comment by lithammer** [comment]
Wouldn't it be better to configure the autopep8 like this instead? https://github.com/hhatto/autopep8#configuration

**Comment by JulianH99** [comment]
@fannheyward Global options work. If I ignore the same arguments on the global coc-settings.json it works fine, but not on the local config

**Comment by JulianH99** [comment]
@lithammer Using pyproject.toml somehow works perfectly
